### PR TITLE
ENH: add consortium standard entrypoint

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Here's an entry point to the Consortium's DataFrame API Standard

It enables dataframe-consuming libraries to just check for a `__dataframe_consortium_standard__` attribute on a dataframe they receive - then, so long as they stick to the spec defined in https://data-apis.org/dataframe-api/draft/index.html, then their code should work the same way, regardless of what the original backing dataframe library was

Use-case: currently, scikit-learn is very keen on using this, as they're not keen on having to depend on pyarrow (which will become required in pandas), and the interchange protocol only goes so far (e.g. a way to convert to ndarray is out-of-scope for that). If we can get this to work there, then other use cases may emerge

The current spec should be enough for scikit-learn, and having this entry point makes it easier to move forwards with development (without monkey-patching / special-casing)

For reference, polars has already merged this: https://github.com/pola-rs/polars/pull/10244

## Maintenance burden on pandas

**None**

I want to be very clear about this: the compat package will be developed, maintained, and tested by the consortium and community of libraries which use it. It is up to consuming libraries to set a minimum version of the dataframe-api-compat package. **No responsibility will land on pandas maintainers**. If any bugs are reported to pandas, then they can (and should) be politely closed.
All this is just an entry point to the Consortium's Standard

Tagging @pandas-dev/pandas-core for visibility. Some people raised objections (maintenance burden, naming) when asked in private, and I think I've addressed the concerns. Anything else?
Thanks 🙌

### CODE REVIEW

**Comment by jbrockmendel** [comment]
No objections here.

**Review by lithomas1 — DISMISSED** [review]
Can you add this to the docs as an optional dependency?
(I assume we'll want a minimum version of the standard at some point)

**Review by lithomas1 — COMMENTED** [review]
This looks good to me now. 
Can you add a whatsnew too in 2.1.0 saying that pandas supports the consortium or something like that too?

File: environment.yml [hunk 1]
```diff
@@ -115,6 +115,7 @@ dependencies:
   - pygments # Code highlighting
 
   - pip:
+      - dataframe-api-compat>=0.1.7
```
Comment (mroeschke) [inline environment.yml:118]: Side question, will this be on conda-forge at some point?

Comment (MarcoGorelli) [reply environment.yml:118]: yup! once I figure that part out :)

File: pyproject.toml [hunk 2]
```diff
@@ -82,11 +82,13 @@ plot = ['matplotlib>=3.6.1']
 output_formatting = ['jinja2>=3.1.2', 'tabulate>=0.8.10']
 clipboard = ['PyQt5>=5.15.6', 'qtpy>=2.2.0']
 compression = ['brotlipy>=0.7.0', 'python-snappy>=0.6.1', 'zstandard>=0.17.0']
+consortium-standard = ['dataframe-api-compat>=0.1.7']
```
Comment (mroeschke) [inline pyproject.toml:85]: Can you add an entry for this in `.github/workflows/package-checks.yml`?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

**Pull request merged**